server/api.js

```diff
@@ -110,21 +110,67 @@ router.post("/energizers", async (req, res) => {
 	}
 });
 
+router.post("/energizers/:id/rate", async (req, res) => {
+	const energizerId = req.params.id;
+	const rating = req.body.value;
+
+	try {
+		// Insert or update rating in the energizer_ratings table
+		await db.query(
+			`INSERT INTO energizer_ratings (energizer_id, rating) 
+				VALUES ($1, $2)`,
+			[energizerId, rating]
+		);
+
+		// Get the average rating of the energizer from the energizer_ratings table
+		const avgRatingResult = await db.query(
+			"SELECT ROUND(AVG(rating), 1) as average_rating FROM energizer_ratings WHERE energizer_id = $1",
+			[energizerId]
+		);
+		const averageRating = avgRatingResult.rows[0].average_rating;
+
+		// Update the average rating of the energizer in the energizers table
+		await db.query(
+			"UPDATE energizer_ratings SET average_rate = $1 WHERE id = $2",
```

Comment:
Why is the average rating being stored in the database? It doesn't seem to be used anyway.

Reply:
on energizers page for each energizer card on the page we display the average rate rate, so I thought if I update the average rate in table those energizer card on energiser page  will show the latest update.

Reply:
But it's being calculated in the database query, there's no need to store it in the database - also if it was stored in the database it should be in the `energizer` table purely for performance reasons (caching).